.travis.yml

```diff
@@ -1,7 +1,7 @@
 sudo: false
 language: java
 install: mvn -B install -DskipTests=true
-script: mvn -B test -P AlsoSlowTests
+script: mvn -B test -P AlsoSlowTests,ExcludeOpenIssueTests
```

Comment:
Wait, does that mean the default run *includes* the open issue tests? So builds fail by default?

Reply:
I'm not sure how best to answer as I'm not 100% sure of how maven/travis etc handle things, but I can say that the profile which skips these is enabled by default.

My best understanding at the moment is that we either need to explicitly pass all profiles, or none of them.

(meaning that `script: mvn -B test` and `script: mvn -B test -P AlsoSlowTests,ExcludeOpenIssueTests` are effectively equivalent)

I might be completely off-base with this though!

Reply:
Alright, well, I think it would have been easier to name both positively: "AlsoSlowTests" and "AlsoOpenIssueTests". It's a bit confusing like this :-) If you agree and have time, do change, otherwise it gets merged like this in a few hours.

Reply:
I'm not sure how to make the logic of this happen as I don't entirely understand how specifying `AlsoSlowTests` allows them to run.

I _think_ what happens is `activeByDefault` applies only when no profile is specified, and is ignored when one or more profiles are applied. 

This would mean that `AlsoSlowTests` should be renamed to `RunAllTests` or `RunAllTests_IncludingThoseExpectedToFail` or similar _IF_ this assumption about how maven works is true.

If renamed this way, it makes more sense why we need to specify <s>`AlsoSlowTests`</s> `RunAllTests` and `ExceptOpenIssueTests` (the full set, followed set subtraction to exclude some).


```xml
        <profile>
            <id>NonSlowTests</id>
            <activation>
                <activeByDefault>true</activeByDefault>
            </activation>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-surefire-plugin</artifactId>
                        <configuration>
                            <excludedGroups>com.github.javaparser.SlowTest</excludedGroups>
                            <parallel>methods</parallel>
                            <threadCount>4</threadCount>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>

        <profile>
            <id>AlsoSlowTests</id>
        </profile>



        <profile>
            <id>ExcludeOpenIssueTests</id>
            <activation>
                <activeByDefault>true</activeByDefault>
            </activation>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-surefire-plugin</artifactId>
                        <configuration>
                            <excludedGroups>com.github.javaparser.OpenIssueTest</excludedGroups>
                            <parallel>methods</parallel>
                            <threadCount>4</threadCount>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
```